libraries/AP_Mount/AP_Mount_SToRM32_serial.cpp

```diff
@@ -38,31 +38,29 @@ void AP_Mount_SToRM32_serial::update()
         // move mount to a "retracted" position.  To-Do: remove support and replace with a relaxed mode?
         case MAV_MOUNT_MODE_RETRACT: {
             const Vector3f &target = _params.retract_angles.get();
-            _angle_rad.roll = ToRad(target.x);
-            _angle_rad.pitch = ToRad(target.y);
-            _angle_rad.yaw = ToRad(target.z);
-            _angle_rad.yaw_is_ef = false;
+            mnt_target.angle_rad.roll = ToRad(target.x);
+            mnt_target.angle_rad.pitch = ToRad(target.y);
+            mnt_target.angle_rad.yaw = ToRad(target.z);
+            mnt_target.target_type = MountTargetType::ANGLE;
+            mnt_target.angle_rad.yaw_is_ef = false;
             break;
         }
 
         // move mount to a neutral position, typically pointing forward
         case MAV_MOUNT_MODE_NEUTRAL: {
             const Vector3f &target = _params.neutral_angles.get();
-            _angle_rad.roll = ToRad(target.x);
-            _angle_rad.pitch = ToRad(target.y);
-            _angle_rad.yaw = ToRad(target.z);
-            _angle_rad.yaw_is_ef = false;
+            mnt_target.angle_rad.roll = ToRad(target.x);
+            mnt_target.angle_rad.pitch = ToRad(target.y);
+            mnt_target.angle_rad.yaw = ToRad(target.z);
+            mnt_target.target_type = MountTargetType::ANGLE;
+            mnt_target.angle_rad.yaw_is_ef = false;
             break;
         }
 
         // point to the angles given by a mavlink message
         case MAV_MOUNT_MODE_MAVLINK_TARGETING:
-            switch (mavt_target.target_type) {
-            case MountTargetType::ANGLE:
-                _angle_rad = mavt_target.angle_rad;
-                break;
-            case MountTargetType::RATE:
-                update_angle_target_from_rate(mavt_target.rate_rads, _angle_rad);
+            if (mnt_target.target_type == MountTargetType::RATE) {
```

Comment:
Let's add a comment here saying that mnt_target should have already been filled in by set_angle_target() or set_rate_target().